packages/wallet-sdk/src/auto-approvals/schemas/policy.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as v from 'valibot';
+
+const AccessLevelSchema = v.union([v.literal('read'), v.literal('mutate'), v.literal('transfer')]);
+
+const BasePermissionSchema = v.object({
+	description: v.string(),
+});
+
+const ObjectTypePermissionSchema = v.object({
+	...BasePermissionSchema.entries,
+	$kind: v.literal('ObjectType'),
+	objectType: v.string(),
+	accessLevel: AccessLevelSchema,
+});
+
+const CoinBalancePermissionSchema = v.object({
+	...BasePermissionSchema.entries,
+	$kind: v.literal('CoinBalance'),
+	coinType: v.string(),
+});
+
+const AnyBalancesPermissionSchema = v.object({
+	...BasePermissionSchema.entries,
+	$kind: v.literal('AnyBalance'),
+});
+
+export type PolicyPermission = v.InferOutput<
+	| typeof CoinBalancePermissionSchema
+	| typeof AnyBalancesPermissionSchema
+	| typeof ObjectTypePermissionSchema
+>;
+
+const AutoApprovalOperationSchema = v.object({
+	id: v.string(),
+	description: v.string(),
+	permissions: v.object({
+		ownedObjects: v.optional(v.array(ObjectTypePermissionSchema)),
+		balances: v.optional(v.array(CoinBalancePermissionSchema)),
+		anyBalance: v.optional(AnyBalancesPermissionSchema),
+	}),
+});
+
+export const AutoApprovalSettingsSchema = v.looseObject({
+	approvedOperations: v.array(v.string()),
```

Comment:
In practice wouldn't you want to keep track of this via something like a deterministic hash of the underlying data so that you can't update the permissions under your feet?